File: frontend/src/model/application/type/application-specifier.js
"use strict";
(function (ApplicationSpecifier) {
    // Katu- ja vihertyöt
    ApplicationSpecifier[ApplicationSpecifier["ASPHALT"] = 0] = "ASPHALT";
    ApplicationSpecifier[ApplicationSpecifier["INDUCTION_LOOP"] = 1] = "INDUCTION_LOOP";
    ApplicationSpecifier[ApplicationSpecifier["COVER_STRUCTURE"] = 2] = "COVER_STRUCTURE";
    ApplicationSpecifier[ApplicationSpecifier["STREET_OR_PARK"] = 3] = "STREET_OR_PARK";
    ApplicationSpecifier[ApplicationSpecifier["PAVEMENT"] = 4] = "PAVEMENT";
    ApplicationSpecifier[ApplicationSpecifier["TRAFFIC_LIGHT"] = 5] = "TRAFFIC_LIGHT";
    ApplicationSpecifier[ApplicationSpecifier["COMMERCIAL_DEVICE"] = 6] = "COMMERCIAL_DEVICE";
    ApplicationSpecifier[ApplicationSpecifier["TRAFFIC_STOP"] = 7] = "TRAFFIC_STOP";
    ApplicationSpecifier[ApplicationSpecifier["BRIDGE"] = 8] = "BRIDGE";
    ApplicationSpecifier[ApplicationSpecifier["OUTDOOR_LIGHTING"] = 9] = "OUTDOOR_LIGHTING";
    // Vesi / viemäri
    ApplicationSpecifier[ApplicationSpecifier["STORM_DRAIN"] = 10] = "STORM_DRAIN";
    ApplicationSpecifier[ApplicationSpecifier["WELL"] = 11] = "WELL";
    ApplicationSpecifier[ApplicationSpecifier["UNDERGROUND_DRAIN"] = 12] = "UNDERGROUND_DRAIN";
    ApplicationSpecifier[ApplicationSpecifier["WATER_PIPE"] = 13] = "WATER_PIPE";
    ApplicationSpecifier[ApplicationSpecifier["DRAIN"] = 14] = "DRAIN";
    // Sähkö
    ApplicationSpecifier[ApplicationSpecifier["DISTRIBUTION_CABINET"] = 15] = "DISTRIBUTION_CABINET";
    ApplicationSpecifier[ApplicationSpecifier["ELECTRICITY_CABLE"] = 16] = "ELECTRICITY_CABLE";
    ApplicationSpecifier[ApplicationSpecifier["ELECTRICITY_WELL"] = 17] = "ELECTRICITY_WELL";
    // Tiedonsiirto
    ApplicationSpecifier[ApplicationSpecifier["DISTRIBUTION_CABINET_OR_PILAR"] = 18] = "DISTRIBUTION_CABINET_OR_PILAR";
    ApplicationSpecifier[ApplicationSpecifier["DATA_CABLE"] = 19] = "DATA_CABLE";
    ApplicationSpecifier[ApplicationSpecifier["DATA_WELL"] = 20] = "DATA_WELL";
    // Lämmitys/viilennys
    ApplicationSpecifier[ApplicationSpecifier["STREET_HEATING"] = 21] = "STREET_HEATING";
    ApplicationSpecifier[ApplicationSpecifier["DISTRICT_HEATING"] = 22] = "DISTRICT_HEATING";
    ApplicationSpecifier[ApplicationSpecifier["DISTRICT_COOLING"] = 23] = "DISTRICT_COOLING";
    // Rakennus
    ApplicationSpecifier[ApplicationSpecifier["GROUND_ROCK_ANCHOR"] = 24] = "GROUND_ROCK_ANCHOR";
    ApplicationSpecifier[ApplicationSpecifier["UNDERGROUND_STRUCTURE"] = 25] = "UNDERGROUND_STRUCTURE";
    ApplicationSpecifier[ApplicationSpecifier["UNDERGROUND_SPACE"] = 26] = "UNDERGROUND_SPACE";
    ApplicationSpecifier[ApplicationSpecifier["BASE_STRUCTURES"] = 27] = "BASE_STRUCTURES";
    ApplicationSpecifier[ApplicationSpecifier["DRILL_PILE"] = 28] = "DRILL_PILE";
    ApplicationSpecifier[ApplicationSpecifier["CONSTRUCTION_EQUIPMENT"] = 29] = "CONSTRUCTION_EQUIPMENT";
    ApplicationSpecifier[ApplicationSpecifier["CONSTRUCTION_PART"] = 30] = "CONSTRUCTION_PART";
    ApplicationSpecifier[ApplicationSpecifier["GROUND_FROST_INSULATION"] = 31] = "GROUND_FROST_INSULATION";
    ApplicationSpecifier[ApplicationSpecifier["SMOKE_HATCH_OR_PIPE"] = 32] = "SMOKE_HATCH_OR_PIPE";
    ApplicationSpecifier[ApplicationSpecifier["STOP_OR_TRANSITION_SLAB"] = 33] = "STOP_OR_TRANSITION_SLAB";
    ApplicationSpecifier[ApplicationSpecifier["SUPPORTING_WALL_OR_PILE"] = 34] = "SUPPORTING_WALL_OR_PILE";
    // Piha
    ApplicationSpecifier[ApplicationSpecifier["FENCE_OR_WALL"] = 35] = "FENCE_OR_WALL";
    ApplicationSpecifier[ApplicationSpecifier["DRIVEWAY"] = 36] = "DRIVEWAY";
    ApplicationSpecifier[ApplicationSpecifier["STAIRS_RAMP"] = 37] = "STAIRS_RAMP";
    ApplicationSpecifier[ApplicationSpecifier["SUPPORTING_WALL_OR_BANK"] = 38] = "SUPPORTING_WALL_OR_BANK";
    // Pohjatutkimus
    ApplicationSpecifier[ApplicationSpecifier["DRILLING"] = 39] = "DRILLING";
    ApplicationSpecifier[ApplicationSpecifier["TEST_HOLE"] = 40] = "TEST_HOLE";
    ApplicationSpecifier[ApplicationSpecifier["GROUND_WATER_PIPE"] = 41] = "GROUND_WATER_PIPE";
    // Muu
    ApplicationSpecifier[ApplicationSpecifier["ABSORBING_SEWAGE_SYSTEM"] = 42] = "ABSORBING_SEWAGE_SYSTEM";
    ApplicationSpecifier[ApplicationSpecifier["GAS_PIPE"] = 43] = "GAS_PIPE";
    ApplicationSpecifier[ApplicationSpecifier["OTHER"] = 44] = "OTHER"; // Muu
})(exports.ApplicationSpecifier || (exports.ApplicationSpecifier = {}));
var ApplicationSpecifier = exports.ApplicationSpecifier;

File: backend/common/src/main/java/fi/hel/allu/common/types/ApplicationSpecifier.java
package fi.hel.allu.common.types;

/**
 * Additional specifiers for certain application kinds
 */
public enum ApplicationSpecifier {
  // Katu- ja vihertyöt
  ASPHALT(ApplicationKind.STREET_AND_GREEN), // Asfaltointityö
  INDUCTION_LOOP(ApplicationKind.STREET_AND_GREEN), // Induktiosilmukka
  COVER_STRUCTURE(ApplicationKind.STREET_AND_GREEN), // Kansisto
  STREET_OR_PARK(ApplicationKind.STREET_AND_GREEN), // Katu tai puisto
  PAVEMENT(ApplicationKind.STREET_AND_GREEN), // Kiveystyö
  TRAFFIC_LIGHT(ApplicationKind.STREET_AND_GREEN), // Liikennevalo
  COMMERCIAL_DEVICE(ApplicationKind.STREET_AND_GREEN), // Mainoslaite
  TRAFFIC_STOP(ApplicationKind.STREET_AND_GREEN), // Pysäkkikatos
  BRIDGE(ApplicationKind.STREET_AND_GREEN), // Silta
  OUTDOOR_LIGHTING(ApplicationKind.STREET_AND_GREEN), // Ulkovalaistus
  // Vesi / viemäri
  STORM_DRAIN(ApplicationKind.WATER_AND_SEWAGE), // Hulevesi
  WELL(ApplicationKind.WATER_AND_SEWAGE), // Kaivo
  UNDERGROUND_DRAIN(ApplicationKind.WATER_AND_SEWAGE), // Salaoja
  WATER_PIPE(ApplicationKind.WATER_AND_SEWAGE), // Vesijohto
  DRAIN(ApplicationKind.WATER_AND_SEWAGE), // Viemäri
  // Sähkö
  DISTRIBUTION_CABINET(ApplicationKind.ELECTRICITY), // Jakokaappi
  ELECTRICITY_CABLE(ApplicationKind.ELECTRICITY), // Kaapeli
  ELECTRICITY_WELL(ApplicationKind.ELECTRICITY), // Kaivo
  // Tiedonsiirto
  DISTRIBUTION_CABINET_OR_PILAR(ApplicationKind.DATA_TRANSFER), // Jakokaappi/-pilari
  DATA_CABLE(ApplicationKind.DATA_TRANSFER), // Kaapeli
  DATA_WELL(ApplicationKind.DATA_TRANSFER), // Kaivo
  // Lämmitys/viilennys
  STREET_HEATING(ApplicationKind.HEATING_COOLING), // Katulämmitys
  DISTRICT_HEATING(ApplicationKind.HEATING_COOLING), // Kaukolämpö
  DISTRICT_COOLING(ApplicationKind.HEATING_COOLING), // Kaukokylmä
  // Rakennus
  GROUND_ROCK_ANCHOR(ApplicationKind.CONSTRUCTION), // Maa- / kallioankkuri
  UNDERGROUND_STRUCTURE(ApplicationKind.CONSTRUCTION), // Maanalainen rakenne
  UNDERGROUND_SPACE(ApplicationKind.CONSTRUCTION), // Maanalainen tila
  BASE_STRUCTURES(ApplicationKind.CONSTRUCTION), // Perusrakenteet
  DRILL_PILE(ApplicationKind.CONSTRUCTION), // Porapaalu
  CONSTRUCTION_EQUIPMENT(ApplicationKind.CONSTRUCTION), // Rakennuksen laite/varuste
  CONSTRUCTION_PART(ApplicationKind.CONSTRUCTION), // Rakennuksen osa
  GROUND_FROST_INSULATION(ApplicationKind.CONSTRUCTION), // Routaeriste
  SMOKE_HATCH_OR_PIPE(ApplicationKind.CONSTRUCTION), // Savunpoistoluukku/-putki, IV-putki
  STOP_OR_TRANSITION_SLAB(ApplicationKind.CONSTRUCTION), // Sulku-/siirtymälaatta
  SUPPORTING_WALL_OR_PILE(ApplicationKind.CONSTRUCTION), // Tukiseinä/-paalu
  // Piha
  FENCE_OR_WALL(ApplicationKind.YARD), // Aita, muuri, penger
  DRIVEWAY(ApplicationKind.YARD), // Kulkutie
  STAIRS_RAMP(ApplicationKind.YARD), // Portaat, luiska tms.
  SUPPORTING_WALL_OR_BANK(ApplicationKind.YARD), // Tukimuuri/-penger, lujitemaamuuri
  // Pohjatutkimus
  DRILLING(ApplicationKind.GEOLOGICAL_SURVEY), // Kairaus
  TEST_HOLE(ApplicationKind.GEOLOGICAL_SURVEY), // Koekuoppa
  GROUND_WATER_PIPE(ApplicationKind.GEOLOGICAL_SURVEY), // Pohjavesiputki
  // Muu
  ABSORBING_SEWAGE_SYSTEM(ApplicationKind.OTHER_CABLE_REPORT), // Imujätejärjestelmä
  GAS_PIPE(ApplicationKind.OTHER_CABLE_REPORT), // Kaasujohto
  OTHER(ApplicationKind.OTHER_CABLE_REPORT); // Muu

  public ApplicationKind getKind() {
    return kind;
  }

  private final ApplicationKind kind;

  private ApplicationSpecifier(ApplicationKind kind) {
    this.kind = kind;
  }

}
